internal/guest/runtime/hcsv2/container.go

```diff
@@ -28,6 +28,13 @@ import (
 	"github.com/Microsoft/hcsshim/internal/protocol/guestresource"
 )
 
+type containerStatus int
+
+const (
+	containerCreating containerStatus = iota
+	containerRunning
```

Comment:
Should this be `containerCreated`, since running implies that `container.Start` was called, and the init process was started?

Reply:
in `CreateContainer` we call into `runc create`, which also starts the container https://github.com/opencontainers/runc/blob/main/create.go#L59

Reply:
but for us, we dont consider the container "started" until we exec the main process, right?
https://github.com/microsoft/hcsshim/blob/master/internal/guest/bridge/bridge_v2.go#L202-L205

Reply:
alright, I missed the part where `runc` passes an action to `startContainer`. You're right and we should track `created` and `running` states separately.

Reply:
I was going to say, it seems confusing to call it running before we actually start the container process. Maybe just `Creating`/`Created` is fine here, or maybe we should have `Running` too? Either way, a comment explaining what exactly the states mean would be helpful.

Reply:
@kevpar , @helsaawy pushed an update, I'd still like to limit it for these two states for now and we can discuss whether we need any other states in the future.